### PR TITLE
Add error classes for symbol delisting errors, closes #270 

### LINKED ISSUE #270: [FEATURE REQUEST] allow supression of "delisted" tickers
when using `yf.download`, have a keyword parameter that will allow us to hide "symbol may be delisted" errors among other errors.
## proposal 
Using `hide_errors=set()` or `suppress_errors=set()` you should check if the error has been specified to be suppressed and then output only if it has not been suppressed

### PR DESCRIPTION
**Pull request by elibroftw**
Breaking changes 
- raise new types of errors
- renames the exceptions that are raised.
- renamed test files
- removed deprecated datetime functions

### CODE REVIEW

**Comment by ValueRaider** [comment]
Can you give a simple scenario that is broken?

**Comment by elibroftw** [comment]
The issue is that the error messages can't be ignored on a per ticker basis easily.

**Comment by elibroftw** [comment]
Cool I'll just maintain my own fork then.

File: requirements.txt [hunk 1]
```diff
@@ -9,3 +9,5 @@ frozendict>=2.3.4
 beautifulsoup4>=4.11.1
 html5lib>=1.1
 peewee>=3.16.2
+requests_cache>=1.0
+requests-ratelimiter>=0.3.1
```
[the remark this answers is not in the document]
Comment (marcofognog) [reply requirements.txt:13]: I guess I ran into the same issue as the author. Those seem to be dev dependencies

**Pull request closed without merging**